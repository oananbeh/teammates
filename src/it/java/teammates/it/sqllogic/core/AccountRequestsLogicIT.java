package teammates.it.sqllogic.core;

import java.time.Instant;
import java.util.UUID;

import org.testng.annotations.Test;

import teammates.common.datatransfer.AccountRequestStatus;
import teammates.common.exception.EntityAlreadyExistsException;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.AccountRequestsLogic;
import teammates.storage.sqlapi.AccountRequestsDb;
import teammates.storage.sqlentity.AccountRequest;

/**
 * SUT: {@link AccountRequestsLogic}.
 */
public class AccountRequestsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private AccountRequestsLogic accountRequestsLogic = AccountRequestsLogic.inst();

    @Test
    public void testGetAccountRequest_nonExistentAccountRequest_returnsNull() {
        UUID id = UUID.randomUUID();
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertNull(actualAccountRequest);
    }

    @Test
    public void testGetAccountRequest_existingAccountRequest_getsSuccessfully() throws InvalidParametersException {
        AccountRequest expectedAccountRequest =
                new AccountRequest("test@gmail.com", "name", "institute", AccountRequestStatus.PENDING, "comments");
        UUID id = expectedAccountRequest.getId();
        accountRequestsLogic.createAccountRequest(expectedAccountRequest);
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertEquals(expectedAccountRequest, actualAccountRequest);
    }

    @Test
    public void testResetAccountRequest()
            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {

        ______TS("success: create account request and update registeredAt field");

        String name = "name lee";
        String email = "email@gmail.com";
        String institute = "institute";
        AccountRequestStatus status = AccountRequestStatus.PENDING;
        String comments = "comments";

        AccountRequest toReset = accountRequestsLogic.createAccountRequest(name, email, institute, status, comments);
        AccountRequestsDb accountRequestsDb = AccountRequestsDb.inst();

        toReset.setRegisteredAt(Instant.now());
        UUID id = toReset.getId();
        toReset = accountRequestsDb.getAccountRequest(id);

        assertNotNull(toReset);
        assertNotNull(toReset.getRegisteredAt());

        ______TS("success: reset account request that already exists");

        AccountRequest resetted = accountRequestsLogic.resetAccountRequest(id);

        assertNull(resetted.getRegisteredAt());

        ______TS("success: test delete account request");

        accountRequestsLogic.deleteAccountRequest(toReset.getId());

        assertNull(accountRequestsLogic.getAccountRequest(toReset.getId()));

        ______TS("failure: reset account request that does not exist");

        assertThrows(EntityDoesNotExistException.class,
                () -> accountRequestsLogic.resetAccountRequest(id));
    }
}
