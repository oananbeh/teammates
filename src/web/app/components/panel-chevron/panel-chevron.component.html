<button type="button" class="chevron btn-course btn btn-primary btn-sm" aria-label="Expand panel" [attr.aria-expanded]=isExpanded>
  <i class="fas fa-chevron-down" [style.color]="chevronColor" *ngIf="!isExpanded"></i>
  <i class="fas fa-chevron-up" [style.color]="chevronColor" *ngIf="isExpanded"></i>
</button>
