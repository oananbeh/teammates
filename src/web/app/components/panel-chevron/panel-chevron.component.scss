button.chevron {
  display: inline-flex;
  vertical-align: middle;
  background-color: inherit;
  border-width: 0;
}

button.chevron:active {
  background-color: inherit;
}
