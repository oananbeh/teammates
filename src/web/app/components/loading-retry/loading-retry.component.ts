import { Component, EventEmitter, Input, Output } from '@angular/core';

/**
 * Retry button for when loading fails.
 */
@Component({
  selector: 'tm-loading-retry',
  templateUrl: './loading-retry.component.html',
  styleUrls: ['./loading-retry.component.scss'],
})
export class LoadingRetryComponent {

  @Input() message: string = '';

  @Input() shouldShowRetry: boolean = false;

  @Input() retryButtonDisabled: boolean = false;

  @Output() retryEvent: EventEmitter<any> = new EventEmitter<any>();

}
