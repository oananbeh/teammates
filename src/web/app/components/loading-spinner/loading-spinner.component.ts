import { Component } from '@angular/core';

/**
 * Loading spinner to show when waiting for request.
 */
@Component({
  selector: 'tm-loading-spinner',
  templateUrl: './loading-spinner.component.html',
  styleUrls: ['./loading-spinner.component.scss'],
})
export class LoadingSpinnerComponent {
  loadingMessage: string = 'Loading...';

  constructor() {
    // After 5 seconds of loading show message to assure user the page is not hanging.
    setTimeout(() => { this.loadingMessage = 'Still loading...'; }, 5000);
  }
}
