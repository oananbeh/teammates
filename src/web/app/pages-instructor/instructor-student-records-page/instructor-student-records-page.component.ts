import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { finalize, map, mergeMap } from 'rxjs/operators';
import { FeedbackResponseCommentService } from '../../../services/feedback-response-comment.service';
import { FeedbackSessionsService } from '../../../services/feedback-sessions.service';
import { InstructorService } from '../../../services/instructor.service';
import { StatusMessageService } from '../../../services/status-message.service';
import { StudentService } from '../../../services/student.service';
import { TableComparatorService } from '../../../services/table-comparator.service';
import {
  FeedbackSession,
  FeedbackSessions,
  Instructor,
  QuestionOutput,
  ResponseOutput,
  SessionResults,
  Student,
} from '../../../types/api-output';
import { Intent } from '../../../types/api-request';
import { SortBy, SortOrder } from '../../../types/sort-properties';
import { CommentToCommentRowModelPipe } from '../../components/comment-box/comment-to-comment-row-model.pipe';
import { CommentsToCommentTableModelPipe } from '../../components/comment-box/comments-to-comment-table-model.pipe';
import { collapseAnim } from '../../components/teammates-common/collapse-anim';
import { ErrorMessageOutput } from '../../error-message-output';
import { InstructorCommentsComponent } from '../instructor-comments.component';

interface SessionTab {
  isCollapsed: boolean;
  feedbackSession: FeedbackSession;
  responsesGivenByStudent: QuestionOutput[];
  responsesReceivedByStudent: QuestionOutput[];
}

/**
 * Instructor student records page.
 */
@Component({
  selector: 'tm-instructor-student-records-page',
  templateUrl: './instructor-student-records-page.component.html',
  styleUrls: ['./instructor-student-records-page.component.scss'],
  animations: [collapseAnim],
})
export class InstructorStudentRecordsPageComponent extends InstructorCommentsComponent implements OnInit {

  courseId: string = '';
  studentName: string = '';
  studentEmail: string = '';
  studentTeam: string = '';
  studentSection: string = '';

  sessionTabs: SessionTab[] = [];

  isStudentLoading: boolean = false;
  hasStudentLoadingFailed: boolean = false;
  isStudentResultsLoading: boolean = false;
  hasStudentResultsLoadingFailed: boolean = false;

  constructor(private route: ActivatedRoute,
              private feedbackSessionsService: FeedbackSessionsService,
              private studentService: StudentService,
              private instructorService: InstructorService,
              private commentsToCommentTableModel: CommentsToCommentTableModelPipe,
              tableComparatorService: TableComparatorService,
              statusMessageService: StatusMessageService,
              commentService: FeedbackResponseCommentService,
              commentToCommentRowModel: CommentToCommentRowModelPipe) {
    super(commentToCommentRowModel, commentService, statusMessageService, tableComparatorService);
  }

  ngOnInit(): void {
    this.route.queryParams.subscribe({
      next: (queryParams: any) => {
        this.courseId = queryParams.courseid;
        this.studentEmail = queryParams.studentemail;

        this.loadStudentRecords();
        this.loadStudentResults();
        this.instructorService.getInstructor({
          courseId: queryParams.courseid,
          intent: Intent.FULL_DETAIL,
        }).subscribe((instructor: Instructor) => {
          this.currInstructorName = instructor.name;
        });
      },
      error: (resp: ErrorMessageOutput) => {
        this.statusMessageService.showErrorToast(resp.error.message);
      },
    });
  }

  /**
   * Loads the student's records based on the given course ID and email.
   */
  loadStudentRecords(): void {
    this.hasStudentLoadingFailed = false;
    this.isStudentLoading = true;
    this.studentService.getStudent(
        this.courseId, this.studentEmail,
    ).pipe(finalize(() => {
      this.isStudentLoading = false;
    })).subscribe({
      next: (resp: Student) => {
        this.studentName = resp.name;
        this.studentTeam = resp.teamName;
        this.studentSection = resp.sectionName;
      },
      error: (resp: ErrorMessageOutput) => {
        this.hasStudentLoadingFailed = true;
        this.statusMessageService.showErrorToast(resp.error.message);
      },
    });
  }

  /**
   * Loads the student's feedback session results based on the given course ID and student name.
   */
  loadStudentResults(): void {
    this.sessionTabs = [];
    this.hasStudentResultsLoadingFailed = false;
    this.isStudentResultsLoading = true;
    this.feedbackSessionsService.getFeedbackSessionsForInstructor(this.courseId).pipe(
        mergeMap((feedbackSessions: FeedbackSessions) => feedbackSessions.feedbackSessions),
        mergeMap((feedbackSession: FeedbackSession) => {
          return this.feedbackSessionsService.getFeedbackSessionResults({
            courseId: this.courseId,
            feedbackSessionName: feedbackSession.feedbackSessionName,
            groupBySection: this.studentSection,
            intent: Intent.FULL_DETAIL,
          }).pipe(map((results: SessionResults) => {
            // sort questions by question number
            results.questions.sort((a: QuestionOutput, b: QuestionOutput) =>
                a.feedbackQuestion.questionNumber - b.feedbackQuestion.questionNumber);
            return { results, feedbackSession };
          }));
        }),
        finalize(() => {
          this.isStudentResultsLoading = false;
        }),
    ).subscribe({
      next: ({ results, feedbackSession }: { results: SessionResults, feedbackSession: FeedbackSession }) => {
        const giverQuestions: QuestionOutput[] = JSON.parse(JSON.stringify(results.questions));
        giverQuestions.forEach((questions: QuestionOutput) => {
          questions.allResponses = questions.allResponses.filter((response: ResponseOutput) =>
              !response.isMissingResponse && response.giverEmail === this.studentEmail);
        });
        const responsesGivenByStudent: QuestionOutput[] =
            giverQuestions.filter((questions: QuestionOutput) => questions.allResponses.length > 0);

        const recipientQuestions: QuestionOutput[] = JSON.parse(JSON.stringify(results.questions));
        recipientQuestions.forEach((questions: QuestionOutput) => {
          questions.allResponses = questions.allResponses.filter((response: ResponseOutput) =>
              !response.isMissingResponse && response.recipientEmail === this.studentEmail);
        });
        const responsesReceivedByStudent: QuestionOutput[] =
            recipientQuestions.filter((questions: QuestionOutput) => questions.allResponses.length > 0);

        this.sessionTabs.push({
          feedbackSession,
          responsesGivenByStudent,
          responsesReceivedByStudent,
          isCollapsed: false,
        });
        results.questions.forEach((questions: QuestionOutput) => this.preprocessComments(questions.allResponses));
      },
      error: (errorMessageOutput: ErrorMessageOutput) => {
        this.hasStudentResultsLoadingFailed = true;
        this.statusMessageService.showErrorToast(errorMessageOutput.error.message);
      },
      complete: () => this.sortFeedbackSessions(),
    });
  }

  /**
   * Preprocesses the comments from instructor.
   *
   * <p>The instructor comment will be moved to map {@code instructorCommentTableModel}. The original
   * instructor comments associated with the response will be deleted.
   */
  preprocessComments(responses: ResponseOutput[]): void {
    const timezone: string = this.sessionTabs[0] ? this.sessionTabs[0].feedbackSession.timeZone : '';
    responses.forEach((response: ResponseOutput) => {
      this.instructorCommentTableModel[response.responseId] =
          this.commentsToCommentTableModel.transform(response.instructorComments, false, timezone);
      this.sortComments(this.instructorCommentTableModel[response.responseId]);
      // clear the original comments for safe as instructorCommentTableModel will become the single point of truth
      response.instructorComments = [];
    });
  }

  /**
   * Sorts the student's feedback sessions according to name
   */
  private sortFeedbackSessions(): void {
    this.sessionTabs.sort((a: SessionTab, b: SessionTab) => {
      return this.tableComparatorService.compare(
          SortBy.SESSION_NAME,
          SortOrder.ASC,
          a.feedbackSession.feedbackSessionName,
          b.feedbackSession.feedbackSessionName);
    });
  }
}
