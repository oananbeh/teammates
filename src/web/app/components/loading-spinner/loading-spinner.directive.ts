import {
  ComponentRef,
  Directive,
  Input,
  TemplateRef,
  ViewContainerRef,
} from '@angular/core';
import { LoadingSpinnerComponent } from './loading-spinner.component';

/**
 * Directive for loading spinner component
 */
@Directive({
  selector: '[tmIsLoading]',
})
export class LoadingSpinnerDirective {
  loadingComponent: ComponentRef<LoadingSpinnerComponent>;

  constructor(private templateRef: TemplateRef<any>,
              private viewContainer: ViewContainerRef) {
    this.loadingComponent = this.viewContainer.createComponent(LoadingSpinnerComponent);
  }

  @Input() set tmIsLoading(loading: boolean) {
    this.viewContainer.clear();

    if (loading) {
      this.loadingComponent = this.viewContainer.createComponent(LoadingSpinnerComponent);
    } else {
      this.viewContainer.createEmbeddedView(this.templateRef);
    }
  }
}
