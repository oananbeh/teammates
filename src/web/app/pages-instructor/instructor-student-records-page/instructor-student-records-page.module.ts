import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { NgbCollapseModule } from '@ng-bootstrap/ng-bootstrap';
import { InstructorStudentRecordsPageComponent } from './instructor-student-records-page.component';
import { CommentToCommentRowModelPipe } from '../../components/comment-box/comment-to-comment-row-model.pipe';
import { CommentsToCommentTableModelPipe } from '../../components/comment-box/comments-to-comment-table-model.pipe';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { PanelChevronModule } from '../../components/panel-chevron/panel-chevron.module';
import {
    GrqRgqViewResponsesModule,
} from '../../components/question-responses/grq-rgq-view-responses/grq-rgq-view-responses.module';

const routes: Routes = [
  {
    path: '',
    component: InstructorStudentRecordsPageComponent,
  },
];

/**
 * Module for instructor student records page.
 */
@NgModule({
  declarations: [
    InstructorStudentRecordsPageComponent,
  ],
  exports: [
    InstructorStudentRecordsPageComponent,
  ],
  imports: [
    CommonModule,
    NgbCollapseModule,
    GrqRgqViewResponsesModule,
    RouterModule.forChild(routes),
    LoadingSpinnerModule,
    LoadingRetryModule,
    PanelChevronModule,
  ],
  providers: [
    CommentToCommentRowModelPipe,
    CommentsToCommentTableModelPipe,
  ],
})
export class InstructorStudentRecordsPageModule { }
