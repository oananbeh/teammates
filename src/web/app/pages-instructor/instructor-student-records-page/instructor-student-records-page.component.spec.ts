import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { of } from 'rxjs';
import { InstructorStudentRecordsPageComponent } from './instructor-student-records-page.component';
import { CommentsToCommentTableModelPipe } from '../../components/comment-box/comments-to-comment-table-model.pipe';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { PanelChevronModule } from '../../components/panel-chevron/panel-chevron.module';
import {
  GrqRgqViewResponsesModule,
} from '../../components/question-responses/grq-rgq-view-responses/grq-rgq-view-responses.module';

describe('InstructorStudentRecordsPageComponent', () => {
  let component: InstructorStudentRecordsPageComponent;
  let fixture: ComponentFixture<InstructorStudentRecordsPageComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [
        InstructorStudentRecordsPageComponent,
      ],
      imports: [
        HttpClientTestingModule,
        RouterTestingModule,
        NgbModule,
        GrqRgqViewResponsesModule,
        LoadingSpinnerModule,
        LoadingRetryModule,
        PanelChevronModule,
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: {
            queryParams: of({ courseid: 'su1337', studentemail: 'punk@punk.com' }),
          },
        },
        CommentsToCommentTableModelPipe,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InstructorStudentRecordsPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with populated fields', () => {
    component.studentName = 'Not John Doe';
    component.courseId = 'su1337';
    component.isStudentLoading = false;
    component.isStudentResultsLoading = false;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when student results are still loading', () => {
    component.studentName = 'John Doe';
    component.courseId = 'CS1111';
    component.isStudentResultsLoading = true;
    component.isStudentLoading = false;

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

});
