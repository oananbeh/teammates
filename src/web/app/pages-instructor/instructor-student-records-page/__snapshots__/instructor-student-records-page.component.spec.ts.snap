// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorStudentRecordsPageComponent should snap when student results are still loading 1`] = `
<tm-instructor-student-records-page
  commentService={[Function FeedbackResponseCommentService]}
  commentToCommentRowModel={[Function CommentToCommentRowModelPipe]}
  commentsToCommentTableModel={[Function CommentsToCommentTableModelPipe]}
  courseId={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasStudentLoadingFailed="false"
  hasStudentResultsLoadingFailed="false"
  instructorCommentTableModel={[Function Object]}
  instructorService={[Function InstructorService]}
  isStudentLoading="false"
  isStudentResultsLoading={[Function Boolean]}
  route={[Function Object]}
  sessionTabs={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentEmail={[Function String]}
  studentName={[Function String]}
  studentSection=""
  studentService={[Function StudentService]}
  studentTeam=""
  tableComparatorService={[Function TableComparatorService]}
>
  <h1
    id="records-header"
  >
    John Doe's Records
    <small
      class="text-muted text-break"
    >
       - CS1111
    </small>
  </h1><h2>
    Records in feedback sessions
  </h2><tm-loading-retry>
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
  </tm-loading-retry>
</tm-instructor-student-records-page>
`;

exports[`InstructorStudentRecordsPageComponent should snap with default fields 1`] = `
<tm-instructor-student-records-page
  commentService={[Function FeedbackResponseCommentService]}
  commentToCommentRowModel={[Function CommentToCommentRowModelPipe]}
  commentsToCommentTableModel={[Function CommentsToCommentTableModelPipe]}
  courseId={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasStudentLoadingFailed="false"
  hasStudentResultsLoadingFailed="false"
  instructorCommentTableModel={[Function Object]}
  instructorService={[Function InstructorService]}
  isStudentLoading={[Function Boolean]}
  isStudentResultsLoading={[Function Boolean]}
  route={[Function Object]}
  sessionTabs={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentEmail={[Function String]}
  studentName=""
  studentSection=""
  studentService={[Function StudentService]}
  studentTeam=""
  tableComparatorService={[Function TableComparatorService]}
>
  <h1
    id="records-header"
  >
    's Records
    <small
      class="text-muted text-break"
    >
       - su1337
    </small>
  </h1><h2>
    Records in feedback sessions
  </h2><tm-loading-retry>
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
  </tm-loading-retry>
</tm-instructor-student-records-page>
`;

exports[`InstructorStudentRecordsPageComponent should snap with populated fields 1`] = `
<tm-instructor-student-records-page
  commentService={[Function FeedbackResponseCommentService]}
  commentToCommentRowModel={[Function CommentToCommentRowModelPipe]}
  commentsToCommentTableModel={[Function CommentsToCommentTableModelPipe]}
  courseId={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasStudentLoadingFailed="false"
  hasStudentResultsLoadingFailed="false"
  instructorCommentTableModel={[Function Object]}
  instructorService={[Function InstructorService]}
  isStudentLoading="false"
  isStudentResultsLoading="false"
  route={[Function Object]}
  sessionTabs={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentEmail={[Function String]}
  studentName={[Function String]}
  studentSection=""
  studentService={[Function StudentService]}
  studentTeam=""
  tableComparatorService={[Function TableComparatorService]}
>
  <h1
    id="records-header"
  >
    Not John Doe's Records
    <small
      class="text-muted text-break"
    >
       - su1337
    </small>
  </h1><h2>
    Records in feedback sessions
  </h2><tm-loading-retry>
    <div>
    </div>
  </tm-loading-retry>
</tm-instructor-student-records-page>
`;
